### PR TITLE
CORE- 48391 - added identityMap data element

### PR DESCRIPTION
**Pull request by ninaceban**
## Description

Added identity map as a data element
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All tests pass and I've made any necessary test changes.
- [ ] I've updated the schema in extension.json or no changes are necessary.

### CODE REVIEW

File: src/view/dataElements/identityMap.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+*/
+
+import "regenerator-runtime"; // needed for some of react-spectrum
+import React from "react";
+import { array, object, string } from "yup";
+import "@react/react-spectrum/Form"; // needed for spectrum form styles
+import render from "../render";
+import ExtensionView from "../components/extensionView";
+import IdentityWrapper from "../components/identityWrapper";
+import getDefaultIdentity from "../utils/getDefaultIdentifier";
```
Comment (jonsnyder) [inline src/view/dataElements/identityMap.jsx:20]: rename this file to match variable name

File: src/view/dataElements/identityMap.jsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            )
+          })
+        )
+      })
+    )
+  })
+  // TestCafe doesn't allow this to be an arrow function because of
+  // how it scopes "this".
+  // eslint-disable-next-line func-names
+  .test("uniqueNamespace", function(settings) {
```
Comment (jonsnyder) [inline src/view/dataElements/identityMap.jsx:103]: If you name the function, then you can avoid the lint error. ie:

.test("uniqueNamespace", function uniqueNamespace(settings) {

File: src/view/utils/getDefaultIdentity.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export default () => {
```
Comment (jonsnyder) [inline src/view/utils/getDefaultIdentity.js:1]: Add Copyright notice. Rename this to getDefaultIdentifier

File: src/view/utils/getDefaultIdentifier.js [hunk 4]
```diff
@@ -13,9 +13,12 @@ governing permissions and limitations under the License.
 export default () => {
```
Comment (jonsnyder) [inline src/view/utils/getDefaultIdentifier.js:13]: rename this to getDefaultIdentity

File: src/view/utils/getDefaultIdentifier.js [hunk 5]
```diff
@@ -13,9 +13,12 @@ governing permissions and limitations under the License.
...
   return {
     namespace: "",
-    id: "",
-    authenticatedState: "",
-    primary: false,
-    hashEnabled: false
+    identities: [
+      {
```
Comment (jonsnyder) [inline src/view/utils/getDefaultIdentifier.js:17]: Import getDefaultIdentifier and call it here.

File: src/view/utils/getDefaultIdentifier.js (hunk 4 above)
Comment (ninaceban) [reply src/view/utils/getDefaultIdentifier.js:13]: thank you

File: extension.json [hunk 6]
```diff
@@ -254,7 +254,24 @@
...
+                "type": "string"
+              },
+              "authenticatedState": {
+                "type": "string",
+                "enum": ["loggedOut", "authenticated", "ambiguous"]
+              },
+              "primary": {
+                "type": "boolean"
+              }
+            }
```
Comment (jonsnyder) [inline extension.json:272]: "additionalProperties": false

Comment (Aaronius) [reply extension.json:272]: Looking at https://github.com/adobe/reactor-extension-alloy/pull/96/files#diff-5711c4c45d1eca09d94a247045264635R91-R94, it seems that `id` and `authenticatedState` are required. If that's the case, they should be marked required here as well.

File: extension.json [hunk 7]
```diff
@@ -248,6 +248,34 @@
...
+          }
+        }
+      },
```
Comment (Aaronius) [inline extension.json:275]: Looking at `https://github.com/adobe/reactor-extension-alloy/pull/96/files#diff-1fa3fffe87cc751f1039a9f91f2cd92aR13`, it seems that the identity map data should be located at `settings.data`, but the schema doesn't look like it reflects this (there's no `data` property in the schema).

Comment (ninaceban) [reply extension.json:275]: yes, thank you, I thought I changed that

File: extension.json (hunk 6 above)
Comment (ninaceban) [reply extension.json:272]: thank you

Comment (ninaceban) [reply extension.json:272]: > Looking at https://github.com/adobe/reactor-extension-alloy/pull/96/files#diff-5711c4c45d1eca09d94a247045264635R91-R94, it seems that `id` and `authenticatedState` are required. If that's the case, they should be marked required here as well.

Actually `authenticatedState` and `primary` are optional with defaults to `ambiguous` and `false`, maybe I should change the Select field to have it default to `ambiguous` and relax validation?

Comment (jfkhoury) [reply extension.json:272]: Yes ambiguous and false should be the default and relaxing the validation makes sense.

File: src/view/utils/getDefaultIdentity.js (hunk 3 above)
Comment (ninaceban) [reply src/view/utils/getDefaultIdentity.js:15]: thank you

**Pull request merged**